src/model_theory/basic.lean

```diff
@@ -192,6 +192,23 @@ lemma encodable.countable_functions [h : encodable (Σl, L.functions l)] :
   L.countable_functions :=
 encodable.countable_functions
 
+@[simp] lemma card_functions_sum :
+  # (Σ n, (L.sum L').functions n) = # ((Σ n, L.functions n) ⊕ (Σ n, L'.functions n)) :=
```

Comment:
Is this in `simp` normal form? If so, it definitely shouldn't be, and there's a lemma about `# (α ⊕ β)` missing.

Reply:
The left-hand side is. The right side is not, but simp will take care of that anyway, and I think this version might be easier to use when one is `rw`-ing by hand, as it doesn't involve any `cardinal.lift`s.

Reply:
I think the problem here is a lack of API. Let's first put `card_eq_card_functions_add_card_relations` in simp-normal form:
```lean
lemma card_eq_card_functions_add_card_relations :
  L.card = cardinal.lift.{v} (cardinal.sum (λ l, #(L.functions l))) +
    cardinal.lift.{u} (cardinal.sum (λ l, #(L.relations l))) :=
by simp [card, symbols]
```
Now, let's add two lemmas currently missing:
```lean
@[simp] theorem sum_add_distrib' {ι} (f g : ι → cardinal) :
  cardinal.sum (λ i, f i + g i) = sum f + sum g :=
sum_add_distrib f g

@[simp] theorem sum_lift {ι : Type u} (f : ι → cardinal.{v}) :
  cardinal.lift.{v'} (cardinal.sum f) = cardinal.sum (λ i, cardinal.lift.{v'} (f i)) :=
equiv.cardinal_eq $ equiv.ulift.trans $ equiv.sigma_congr_right $ λ a, nonempty.some $
  by rw [←lift_mk_eq, mk_out, mk_out, lift_lift]
```
Now, your `cardinal_{functions/relations}_sum` lemmas can be solved directly by `simp`. Moreover, the lemma that you actually want still has essentially the same proof:
```lean
@[simp] lemma card_sum :
  (L.sum L').card = cardinal.lift.{max u' v'} L.card + cardinal.lift.{max u v} L'.card :=
begin
  simp only [card_eq_card_functions_add_card_relations, card_functions_sum, card_relations_sum,
    sum_add_distrib', lift_add, sum_lift, lift_lift],
  rw [add_assoc, ←add_assoc (cardinal.sum (λ i, (# (L'.functions i)).lift)),
    add_comm (cardinal.sum (λ i, (# (L'.functions i)).lift))],
  repeat { rw add_assoc }
end
```
I'll add the missing cardinal lemmas in another PR.

Reply:
#13838

Reply:
@awainverse :arrow_up: has now been merged, can you use that to implement some of @vihdzp's suggestions?

Reply:
The suggestions have now been implemented.